src/unix/tcp.c

```diff
@@ -272,10 +274,27 @@ int uv_tcp_listen(uv_tcp_t* tcp, int backlog, uv_connection_cb cb) {
   if (err)
     return err;
 
+#ifdef __MVS__
+  /* on zOS the listen call does not bind automatically 
+     if the socket is unbound. Hence the manual binding to 
+     an arbitrary port is required to be done manually 
+  */
+
+  if (!(tcp->flags & UV_HANDLE_BOUND))
+  {
+    memset(&saddr, 0, sizeof(saddr));
+    saddr.sin_family = AF_INET;
```

Comment:
Only if the socket wasn't created previously. The passed domain is used as the default if there is no socket created previously.